node/src/command.rs

```diff
@@ -207,31 +311,51 @@ pub fn run() -> sc_cli::Result<()> {
         }
         Some(Subcommand::Revert(cmd)) => {
             let runner = cli.create_runner(cmd)?;
-            runner.async_run(|config| {
-                let PartialComponents { client, task_manager, backend, .. } =
-                    new_partial::<RuntimeApi, ExecutorDispatch>(&config)?;
+            runner.async_run(|mut config| {
+                let (client, backend, _, task_manager) = new_chain_ops(&mut config)?;
 
-                let aux_revert = Box::new(move |client, _, blocks| {
-                    sc_finality_grandpa::revert(client, blocks)?;
-                    Ok(())
-                });
-
-                Ok((cmd.run(client, backend, Some(aux_revert)), task_manager))
+                Ok((cmd.run(client, backend, None), task_manager))
             })
         }
         #[cfg(feature = "try-runtime")]
         Some(Subcommand::TryRuntime(cmd)) => {
             let runner = cli.create_runner(cmd)?;
-            runner.async_run(|config| {
-                // we don't need any of the components of new_partial, just a runtime, or a task
-                // manager to do `async_run`.
-                let registry = config.prometheus_config.as_ref().map(|cfg| &cfg.registry);
-                let task_manager =
-                    sc_service::TaskManager::new(config.tokio_handle.clone(), registry)
-                        .map_err(|e| sc_cli::Error::Service(sc_service::Error::Prometheus(e)))?;
-
-                Ok((cmd.run::<Block, ExecutorDispatch>(config), task_manager))
-            })
+            let chain_spec = &runner.config().chain_spec;
+
+            match chain_spec {
+                #[cfg(feature = "with-zeitgeist-runtime")]
+                spec if spec.is_zeitgeist() => {
```

Comment:
what happens if accidentally specify spec which is not zeitgiest() with feature "with-zeitgeist-runtime"? 
has it handled some other place in code? Or that can never happen?

Reply:
It will try to use the Battery Station runtime. If the node was built without the Battery Station runtime, it will return an error.